scaletest/dashboard/run.go

```diff
@@ -83,49 +85,12 @@ func (*Runner) Cleanup(_ context.Context, _ string) error {
 	return nil
 }
 
-func (r *Runner) do(ctx context.Context, act RollTableEntry, p *Params) {
-	select {
-	case <-ctx.Done():
-		r.cfg.Logger.Info(ctx, "context done, stopping")
-		return
-	default:
-		var errored bool
-		cancelCtx, cancel := context.WithTimeout(ctx, r.cfg.MaxWait)
-		defer cancel()
-		start := time.Now()
-		err := act.Fn(cancelCtx, p)
-		cancel()
-		elapsed := time.Since(start)
-		if err != nil {
-			errored = true
-			r.cfg.Logger.Error( //nolint:gocritic
-				ctx, "action failed",
-				slog.Error(err),
-				slog.F("action", act.Label),
-				slog.F("elapsed", elapsed),
-			)
-		} else {
-			r.cfg.Logger.Info(ctx, "completed successfully",
-				slog.F("action", act.Label),
-				slog.F("elapsed", elapsed),
-			)
-		}
-		codeLabel := "200"
-		if apiErr, ok := codersdk.AsError(err); ok {
-			codeLabel = fmt.Sprintf("%d", apiErr.StatusCode())
-		} else if xerrors.Is(err, context.Canceled) {
-			codeLabel = "timeout"
-		}
-		r.metrics.ObserveDuration(act.Label, elapsed)
-		r.metrics.IncStatuses(act.Label, codeLabel)
-		if errored {
-			r.metrics.IncErrors(act.Label)
-		}
+func (r *Runner) randWait() time.Duration {
+	var wait time.Duration
+	if r.cfg.MaxWait > r.cfg.MinWait {
```

Comment:
nit: maybe you should implement some prediction to fail fast instead of having this condition

Reply:
Can do this in `Config.Validate()` instead.

Reply:
SGTM!